flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/ApplicationObserver.java

```diff
@@ -70,23 +70,28 @@ protected Optional<String> updateJobStatus(
     }
 
     @Override
-    public void observeIfClusterReady(
-            FlinkDeployment flinkApp, Context context, Configuration lastValidatedConfig) {
+    protected boolean observeFlinkCluster(
+            FlinkDeployment flinkApp, Context context, Configuration deployedConfig) {
+
+        JobStatus jobStatus = flinkApp.getStatus().getJobStatus();
+
         boolean jobFound =
                 jobStatusObserver.observe(
-                        flinkApp.getStatus().getJobStatus(),
-                        lastValidatedConfig,
-                        new ApplicationObserverContext(flinkApp, context, lastValidatedConfig));
+                        jobStatus,
+                        deployedConfig,
+                        new ApplicationObserverContext(flinkApp, context, deployedConfig));
         if (jobFound) {
             savepointObserver
-                    .observe(
-                            flinkApp.getStatus().getJobStatus().getSavepointInfo(),
-                            flinkApp.getStatus().getJobStatus().getJobId(),
-                            lastValidatedConfig)
+                    .observe(jobStatus.getSavepointInfo(), jobStatus.getJobId(), deployedConfig)
                     .ifPresent(
                             error ->
                                     ReconciliationUtils.updateForReconciliationError(
                                             flinkApp, error));
         }
+        return isJobReady(jobStatus);
+    }
+
+    private boolean isJobReady(JobStatus jobStatus) {
+        return org.apache.flink.api.common.JobStatus.RUNNING.name().equals(jobStatus.getState());
```

Comment:
Can we add a TODO here? RUNNING doesn't mean that the job is executing as expected, even a job that flip flops may intermittently have RUNNING status.

Reply:
yes I will add a Jira for this before merging